### DISCUSSION TITLE
MikroORM does not detect TS config in Nest

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2454] [Q&A] [answered]
I am just trying out the CLI, but it does not detect my `{repo_root}/src/mikro-orm.config.ts` config.

Run in the repo root folder, the output of `npx mikro-orm debug` is as follows:

```bash
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 5.0.0-dev.510+6f23fb07e
   - node 16.13.0
   - typescript 4.5.2
 - package.json found
 - searched config paths:
   - {repo_root}/mikro-orm.config.js (not found)
- configuration not found (MikroORM config file not found in ['./mikro-orm.config.js'])
```

Running in `{repo_root}/src` folder does not even find `package.json`, which I’d consider a bug (IMHO, the CLI tool should be able to detect the nearest `package.json`, starting in the CWD and if it’s not found, look in the parent folders), but that is a separate issue. However, even then it does not detect the config file.

I have `ts-node` installed locally.

Here is the config I use:

<details>
  <summary><code>mikro-orm.config.ts</code></summary>

```ts
import {Logger} from '@nestjs/common'
import {Options} from '@mikro-orm/core'
import {SqlHighlighter} from '@mikro-orm/sql-highlighter'
import {TsMorphMetadataProvider} from '@mikro-orm/reflection'
import {ConfigService} from '@nestjs/config'

const logger = new Logger('MikroORM')
const configService = new ConfigService

const config = {
  autoLoadEntities: true,
  dbName: configService.get<string>('PSQL_DB_NAME_CS'),
  type: 'postgresql',
  host: configService.get<string>('PSQL_HOST', 'localhost'),
  port: configService.get<number>('PSQL_PORT', 5432),
  highlighter: new SqlHighlighter(),
  debug: false,
  logger: logger.log.bind(logger),
  username: configService.get<string>('PSQL_USER'),
  password: configService.get<string>('PSQL_PASS'),
  synchronize: true,
  metadataProvider: TsMorphMetadataProvider,
  migrations: {
    tableName: 'mikro_orm_migrations',
    path: './migrations',
    pattern: /^[\w-]+\d+\.ts$/,
    transactional: true,
    disableForeignKeys: true,
    allOrNothing: true,
    dropTables: true,
    safe: true,
    emit: 'ts'
  }
} as Options

if (configService.get<string>('LOG_LEVEL', 'warn') === 'debug' || configService.get<string>('LOG_LEVEL', 'warn') === 'trace') {
  config.debug = true
}

export default config
```

</details>

---

**Update:** Can this be related to custom env configuration? I am using `ConfigModule.forRoot({load: [config]})` in the Nest root module, where `config` is an import from a simple TS file (see [this] for more information).

**Comment by B4nan** [comment, marked as answer]
You need to enable ts-node support in package.json via `useTsNode`

https://mikro-orm.io/docs/installation#setting-up-the-commandline-tool

Also it already does lookup the parent folders for package.json in v5

https://github.com/mikro-orm/mikro-orm/blob/master/tests/features/cli/CLIHelper.test.ts#L340

**Reply by tukusejssirs** [reply]
How stupid I am! :man_facepalming: I’ve read the docs, and I’ve thought I’ve already added that to `package.json`, but indeed I forget to do so. Thanks!

**Reply by tukusejssirs** [reply]
But now I get a different error: `- configuration not found (No database specified, please fill in `dbName` or `clientUrl` option)`. Is this related to the fact that I use a custom global configuration (see the updated OP)?

**Reply by B4nan** [reply]
idk, try putting console.logs to the config file, maybe you need to register dotenv or something like that?

**Reply by B4nan** [reply]
(dotenv is registered in the ORM automatically, but might happen too late for your use case)

**Reply by tukusejssirs** [reply]
Right. I loaded both `ConfigService` and `config` in the MikroORM config. Now it works, but it feels like a hack and it is not very DRY.

**Reply by tukusejssirs** [reply]
> (dotenv is registered in the ORM automatically, but might happen too late for your use case)

Well, even though I use `dotenv` (via `@nestjs/config`) currently, I don’t use any `.env` file, as I need to port a PHP+AJAX+JS+Node.js+BASH+Perl+Python hybrid to Node.js/TS/Nest and the config is currently in a PHP file (using `putenv()`), so I need to parse that file and create an object from it. Therefore my assumption is proved. Thanks again! :wink:
